otsclient/cmds.py

```diff
@@ -57,18 +57,34 @@ def create_timestamp(timestamp, calendar_urls, args):
     if setup_bitcoin:
         proxy = setup_bitcoin()
 
-        unfunded_tx = CTransaction([], [CTxOut(0, CScript([OP_RETURN, timestamp.msg]))])
-        r = proxy.fundrawtransaction(unfunded_tx)  # FIXME: handle errors
-        funded_tx = r['tx']
+        txid = None
+        if args.txid:
+            logging.debug("Continue with existing transaction")
+            txid = bytes.fromhex(args.txid)[::-1]
+        else:
+            logging.debug("Call fundrawtransaction for OP_RETURN %s", timestamp.msg.hex())
+            unfunded_tx = CTransaction([], [CTxOut(0, CScript([OP_RETURN, timestamp.msg]))])
+
+            options = {}
+            if args.fee_rate:
+                options['fee_rate'] = args.fee_rate
+
+            r = proxy.fundrawtransaction(unfunded_tx, options)  # FIXME: handle errors
+            funded_tx = r['tx']
 
-        r = proxy.signrawtransaction(funded_tx)
-        assert r['complete']
-        signed_tx = r['tx']
+            logging.debug("Call signrawtransactionwithwallet %s", funded_tx.serialize().hex())
+            r = proxy.signrawtransactionwithwallet(funded_tx)
+            assert r['complete']
+            signed_tx = r['tx']
 
-        txid = proxy.sendrawtransaction(signed_tx)
-        logging.info('Sent timestamp tx')
+            logging.debug("Call sendrawtransaction %s", signed_tx.serialize().hex())
+            txid = proxy.sendrawtransaction(signed_tx)
+            logging.info('Sent timestamp tx')
 
         blockhash = None
+
+        logging.info('Waiting for confirmation. This can be interupted and resumed with:\nots stamp --nonce=%s --txid=%s', nonce.hex(), txid[::-1].hex())
```

Comment:
This doesn't actually work as intended. You need to run `ots stamp -b --nonce=x --txid=x`. If the `-b` is left out stamping happens normally, ignoring the `nonce` and `txid` arguments.

Maybe we want `--nonce` and `--txid` to force `-b` on?

Reply:
Another issue we have right now is that if you use the `txid` and `nonce` options *after* a block has confirmed, ots fails to find the new block.

Maybe we could query the wallet for the transaction? Or maybe we also need a block height?

If we have all three options, we probably want to do it via a single --resume option that can find the tx even if the txid changes.